src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py

```diff
@@ -193,11 +199,78 @@ def flexible_server_update_custom_func(cmd, instance,
+            raise CLIError('Argument Error. If you pass --tier, --sku_name is a mandatory parameter and vice-versa.')
+
+        if tier is None and sku_name is None:
+            iops = _determine_iops(storage_mb, iops_info, iops, tier, sku_name)
+        else:
+            new_sku_rank = sku_rank[sku_name]
+            old_sku_rank = sku_rank[instance.sku.name]
+            supplied_iops = iops
+            max_allowed_iops_new_sku = iops_info[tier][sku_name]
+            default_iops = 100
+            if storage_mb is None:
+                free_iops = instance.storage_profile.storage_mb * 3
+            else:
+                free_iops = storage_mb * 3
+            # Downgrading SKU
+            if new_sku_rank < old_sku_rank:
+                if supplied_iops > max_allowed_iops_new_sku:
+                    iops = max_allowed_iops_new_sku
+                    logger.warning('The max IOPS for your sku is %s. Provisioning the server with %s...', iops, iops)
+                elif supplied_iops < default_iops:
+                    if free_iops < default_iops:
+                        iops = default_iops
+                        logger.warning('The min IOPS is %s. Provisioning the server with %s...', default_iops,
+                                       default_iops)
+                    else:
+                        iops = min(max_allowed_iops_new_sku, free_iops)
+                        logger.warning('Provisioning the server with %s IOPS...', iops)
+            else:  # Upgrading SKU
+                if supplied_iops > max_allowed_iops_new_sku:
+                    iops = max_allowed_iops_new_sku
+                    logger.warning(
+                        'The max IOPS for your sku is %s. Provisioning the server with %s...', iops, iops)
+                elif supplied_iops <= max_allowed_iops_new_sku:
+                    iops = max(supplied_iops, min(free_iops, max_allowed_iops_new_sku))
+                    if iops != supplied_iops:
+                        logger.warning('Provisioning the server with %s IOPS...', iops)
+                elif supplied_iops < default_iops:
+                    if free_iops < default_iops:
+                        iops = default_iops
+                        logger.warning(
+                            'The min IOPS is %s. Provisioning the server with %s...', default_iops, default_iops)
+                    else:
+                        iops = min(max_allowed_iops_new_sku, free_iops)
+                        logger.warning('Provisioning the server with %s IOPS...', iops)
+            instance.sku.name = sku_name
+            instance.sku.tier = tier
+        instance.storage_profile.storage_iops = iops
+
+    # pylint: disable=too-many-boolean-expressions
+    if (iops is None and tier is None and sku_name) or (iops is None and sku_name is None and tier):
+        raise CLIError('Argument Error. If you pass --tier, --sku_name is a mandatory parameter and vice-versa.')
```

Comment:
From previous code, there seems no restriction that `--sku-name` is a mandatory when passing `--tier`. An existing example shows like this:
https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/rdbms/_helptext_mysql.py#L270-L272

Is this breaking change expected?

Reply:
This wont be a breaking change. Previously - it would have anyways failed without a proper error message like ' Deployment Failed. Correlation ID .....' if one of the params were missing. Now, with this change, it will still continue to fail, but with a meaningful error message and the reason as to why it is failing.

Reply:
I didn't know that two params are required not only for create, but also for update. In that case, could you correct the example? I can still see this in help:
```
Examples
    Update a flexible server's sku, using local context for server and resource group.
        az mysql flexible-server update --sku-name Standard_D4s_v3
```

Reply:
Fixed this help text. Thanks for pointing it out!